### PR TITLE
Use newer version of statsd client

### PR DESCRIPTION
**Pull request by mar-kolya**
This should fix 'index out of bounds' issue that happened in older client

### CODE REVIEW

File: dd-trace-ot/dd-trace-ot.gradle [hunk 1]
```diff
@@ -39,13 +39,13 @@ dependencies {
...
 
-  compile group: 'com.datadoghq', name: 'java-dogstatsd-client', version: '2.1.1'
+  compile group: 'com.datadoghq', name: 'java-dogstatsd-client', version: "${versions.dogstatsd}"
 
   compile deps.slf4j
   compile deps.okhttp
   compile group: 'org.msgpack', name: 'msgpack-core', version: '0.8.20'
   compile group: 'com.squareup.moshi', name: 'moshi', version: '1.9.2'
-  compile group: 'com.github.jnr', name: 'jnr-unixsocket', version: '0.23'
+  compile group: 'com.github.jnr', name: 'jnr-unixsocket', version: "${versions.jnr_unixsocket}"
```
Comment (jbachorik) [inline dd-trace-ot/dd-trace-ot.gradle:48]: Is this related to the statsd client?

Comment (mar-kolya) [reply dd-trace-ot/dd-trace-ot.gradle:48]: Indirectly: statsd depends on this and we separately depend on this. And the way we package things means we need to depend on the same thing.

Comment (jbachorik) [reply dd-trace-ot/dd-trace-ot.gradle:48]: Ok
